module-4/backend/week_two.md

```diff
@@ -0,0 +1,45 @@
+## Week Two - Module 4 Recap
+
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+
+Note: When you're done, submit a PR. 
+
+1. What is Webpack and why is it useful?
+- Wepback is a package for node projects that is similar to the asset pipeline in Rails. It looks at your asset files in your project (JS, CSS etc) and packages them up into one file per asset (so one js, one css) for production. There is a difference between what we as developers want/use to make our lives easier - multiple files, white space etc. and what the broswer wants - one file, no white space and webpack takes what we have written and makes it easier for the browser to read it. 
+
+2. When do you want to use event delegation?
+- Event delegation is good in cases where you might have memory leak - you tied an event listener to something that is no longer in the DOM and so the program is "listening" for something that will never happen and wasting memory use on it. Event delegation ties the listener to the parent object that then analyzes the event to see if it matches exactly what you are looking for. 
```

Comment:
it's also useful for when you need to bind events to elements that are not on the page on page load